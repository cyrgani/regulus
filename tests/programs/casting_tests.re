assert_eq("2", string(2)),
assert_eq("2", string("2")),
assert_eq(2, int("2")),
assert_eq(2, int(string(2))),
assert_eq(true, bool(1)),
assert_eq(string(int(bool(int("2")))), "1"),
assert_eq(bool(2), true),
assert_eq(bool(-1), true),
assert_eq(bool(null), false),
__builtin_print_catch(bool("true")),
__builtin_print_catch(bool("false")),
__builtin_print_catch(bool("")),
__builtin_print_catch(int("a")),
assert_eq(string(true), "true"),
=(x, 2),
assert_eq(string(x), "2"),
# TODO: extend this drastically
