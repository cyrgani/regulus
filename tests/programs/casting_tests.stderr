TypeError: cannot cast true to bool
at <stl:casting>:17:22
at <stl:casting>:14:19
at <stl:casting>:11:15
at <stl:casting>:8:11
at <stl:casting>:6:17
at programs/casting_tests.re:10:27
at programs/casting_tests.re:10:22
TypeError: cannot cast false to bool
at <stl:casting>:17:22
at <stl:casting>:14:19
at <stl:casting>:11:15
at <stl:casting>:8:11
at <stl:casting>:6:17
at programs/casting_tests.re:11:27
at programs/casting_tests.re:11:22
TypeError: cannot cast  to bool
at <stl:casting>:17:22
at <stl:casting>:14:19
at <stl:casting>:11:15
at <stl:casting>:8:11
at <stl:casting>:6:17
at programs/casting_tests.re:12:27
at programs/casting_tests.re:12:22
TypeError: Unable to cast a to int
at programs/casting_tests.re:13:26
at programs/casting_tests.re:13:22
